lib/ansible/module_utils/network/meraki/meraki.py

```diff
@@ -0,0 +1,408 @@
+        self.function = function
+
+        if module.params['auth_key'] is None:
+            module.fail_json(msg='Meraki API key not specified')
+        else:
+            self.headers = {'Content-Type': 'application/json',
+                            'X-Cisco-Meraki-API-Key': module.params['auth_key'],
+                            }
+
+        # error output
+        self.error = dict(code=None, text=None)
+
+        # normal output
+        self.existing = None
+
+        # info output
+        self.config = dict()
+        self.original = None
+        self.proposed = dict()
+        self.merged = None
+
+        # debug output
+        self.filter_string = ''
+        self.method = None
+        self.path = None
+        self.response = None
+        self.status = None
+        self.url = None
+
+        '''
+        If URLs need to be modified or added for specific purposes, use .update() on the url_catalog dictionary
+        '''
+        self.get_urls = {'organizations': '/organizations',
+                         'networks': '/organizations/{org_id}/networks',
+                         'admins': '/organizations/{org_id}/admins',
+                         'configTemplates': '/organizations/{org_id}/configTemplates',
+                         'samlRoles': '/organizations/{org_id}/samlRoles',
+                         'ssids': '/networks/{net_id}/ssids',
+                         'groupPolicies': '/networks/{net_id}/groupPolicies',
+                         'staticRoutes': '/networks/{net_id}/staticRoutes',
+                         'vlans': '/networks/{net_id}/vlans',
+                         'devices': '/networks/{net_id}/devices',
+                         }
+
+        self.get_one_urls = {'organizations': '/organizations/{org_id}',
+                             'networks': '/networks/{net_id}',
+                             }
+
+        # Module should add URLs which are required by the module
+        self.url_catalog = {'get_all': self.get_urls,
+                            'get_one': self.get_one_urls,
+                            'create': None,
+                            'update': None,
+                            'delete': None,
+                            'misc': None,
+                            }
+
+        if self.module._debug or self.params['output_level'] == 'debug':
+            self.module.warn('Enable debug output because ANSIBLE_DEBUG was set or output_level is set to debug.')
+
+        # TODO: This needs to be tested
+        self.module.required_if = [('state', 'present', ['name']),
+                                   ('state', 'absent', ['name']),
+                                   ]
+        self.module.mutually_exclusive = [('org_id', 'org_name'),
+                                          ]
+        self.modifiable_methods = ['POST', 'PUT', 'DELETE']
+
+        if self.params['auth_key'] is None:
+            try:
+                self.params['auth_key'] = os.environ['MERAKI_KEY']
```

Comment:
Didn't we already took care of this a little bit up ?

Reply:
We moved it into the module utility.

Reply:
Fixed.

Reply:
I still don't get it. This was taken care of also in the arg_spec. So I would have left line 60 as it was and removed this.